ccp/src/cu/cu_prover.rs

```diff
@@ -161,22 +172,35 @@ impl CUProver {
         &'threads mut self,
         cache: CacheHandle,
         dataset: DatasetHandle,
+        cancel_token: CancellationToken,
     ) -> CUResult<()> {
         use futures::FutureExt;
 
-        let threads_number = self.threads.len() as u64;
         let dataset_size = dataset.items_count();
+        let threads_number = self.threads.len() as u64;
 
         let closure = |thread_id: usize, thread: &'threads mut ProvingThreadAsync| {
             let thread_id = thread_id as u64;
 
             let start_item = (dataset_size * thread_id) / threads_number;
             let next_start_item = (dataset_size * (thread_id + 1)) / threads_number;
-            let items_count = next_start_item - start_item;
 
-            thread
-                .initialize_dataset(cache.clone(), dataset.clone(), start_item, items_count)
-                .boxed()
+            let chunk_generator =
+                ChunkGenerator::new(start_item, next_start_item, DATASET_CHUNK_SIZE);
```

Comment:
let's make `DATASET_CHUNK_SIZE` not a constant, but parameter in a config

Reply:
Done.